kinto/core/storage/postgresql/client.py

```diff
@@ -75,29 +74,28 @@ def create_from_config(config, prefix='', with_transaction=True):
     from sqlalchemy.orm import sessionmaker, scoped_session
 
     settings = {**config.get_settings()}
+
     # Custom Kinto settings, unsupported by SQLAlchemy.
-    settings.pop(prefix + 'backend', None)
-    settings.pop(prefix + 'max_fetch_size', None)
-    settings.pop(prefix + 'max_size_bytes', None)
-    settings.pop(prefix + 'prefix', None)
-    settings.pop(prefix + 'strict_json', None)
-    settings.pop(prefix + 'hosts', None)
-    transaction_per_request = with_transaction and settings.pop('transaction_per_request', False)
-
-    url = settings[prefix + 'url']
+    black_list = create_black_list(prefix)
+    settings_key_list = settings.keys()
+    white_list = list(filter(lambda x: x not in black_list, settings_key_list))
+    filtered_settings = {k: v for k, v in settings.items() if k in white_list}
```

Comment:
a more idiomatic (and shorter) way of writing these 3 lines would be:

`filtered_settings = {k: v for k, v in settings.items() if k not in EXCLUDE_SETTINGS}`